File: src/main/java/start/HelloThreadMain.java
package start;

public class HelloThreadMain {
    public static void main(String[] args) {

        System.out.println(Thread.currentThread().getName() + " - main() start");

        HelloThread thread = new HelloThread();
        System.out.println(Thread.currentThread().getName() + " - hello thread 호출 전");
        thread.start();
        System.out.println(Thread.currentThread().getName()+ " hello thread 호출 후");

        System.out.println(Thread.currentThread().getName() + " - main() end");
    }

}

File: src/main/kotlin/start/HelloThreadMain.kt
package org.example.start

class HelloThreadMain {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            println("${Thread.currentThread().name} - main() start")
            val thread = HelloThread()
            println("${Thread.currentThread().name} - hello thread 호출 전")
            thread.start()
            println("${Thread.currentThread().name} - hello thread 호출 후")
            println("${Thread.currentThread().name} - main() end")
        }
    }

}

fun main() {

    println("${Thread.currentThread().name} - main() start")
    val thread = HelloThread()
    println("${Thread.currentThread().name} - hello thread 호출 전")
    thread.start()
    println("${Thread.currentThread().name} - hello thread 호출 후")
    println("${Thread.currentThread().name} - main() end")
}